src/main/java/de/tum/in/www1/artemis/repository/SubmissionRepository.java

```diff
@@ -555,4 +555,41 @@ SELECT MAX(s2.submissionDate)
     default Submission findByIdElseThrow(long submissionId) {
         return findById(submissionId).orElseThrow(() -> new EntityNotFoundException("Submission", submissionId));
     }
+
+    /**
+     * Gets all unassessed Quiz Submissions for the given exam
+     *
+     * @param examId the ID of the exam
+     * @return boolean indicating if there are unassessed Quiz Submission
+     */
+    @Query("""
+                SELECT COUNT(p.exercise) > 0
+                FROM StudentParticipation p
+                    JOIN p.submissions s
+                    LEFT JOIN s.results r
+                WHERE p.exercise.exerciseGroup.exam.id = :examId
+                    AND p.testRun IS FALSE
+                    AND TYPE(s) = QuizSubmission
+                    AND s.submitted IS TRUE
+                    AND r.id IS NULL
+            """)
+    boolean existsUnassessedQuizzesByExamId(@Param("examId") long examId);
+
+    /**
+     * Checks if unsubmitted text and modeling submissions exist for the exam with the given id
+     *
+     * @param examId the ID of the exam
+     * @return boolean indicating if there are unsubmitted text and modelling submissions
+     */
+    @Query("""
+            SELECT COUNT(p.exercise) > 0
+            FROM StudentParticipation p
+                JOIN p.submissions s
+            WHERE p.exercise.exerciseGroup.exam.id = :examId
+                AND p.testRun IS FALSE
+                AND TYPE(s) IN (TextSubmission, ModelingSubmission)
+                AND s.submitted IS NULL
```

Comment:
```suggestion
                AND (s.submitted IS NULL OR s.submitted IS FALSE)
```
Do we need that here?

Reply:
I was not sure because in the table it is either 1 or null. But I can add this just to be sure